Documentation/Input/CreateDataProvider.md

```diff
@@ -0,0 +1,202 @@
+# Creating an input system data provider
+
+The Mixed Reality Toolkit input system is an extensible system for enabling input device support. To add support for a new hardware platform,
+a custom input data provider may be required.
+
+This article describes how to create custom data providers, also called device managers, for the input system. The example code shown here is
+from the [`WindowsMixedRealityDeviceManager`](xref:Microsoft.MixedReality.Toolkit.WindowsMixedReality.Input.WindowsMixedRealityDeviceManager).
+
+> The complete code used in this example can be found in the MixedRealityToolkit.Providers\WindowsMixedReality folder.
+
+## Namespace and folder structure
+
+> [!Important]
+> If an input system data provider is being submitted to the [Mixed Reality Toolkit repository](https://github.com/Microsoft/MixedRealityToolkit-Unity), the
```

Comment:
What are the requirements for being accepted?

Reply:
Good questions!

Reply:
adding the recommendation for the ThirdParty folder. I really like that idea.

Reply:
updated in latest commit